crates/forge/src/init.rs

```diff
@@ -1,223 +1,24 @@
-
-    Ok(())
-}
-
-fn add_test_script(document: &mut DocumentMut) {
-    let mut test = Table::new();
-
-    test.insert("test", value("snforge test"));
-    document.insert("scripts", Item::Table(test));
-}
-
-fn add_target_to_toml(document: &mut DocumentMut) {
-    let mut array_of_tables = ArrayOfTables::new();
-    let mut sierra = Table::new();
-    let mut contract = Table::new();
-    contract.set_implicit(true);
-
-    sierra.insert("sierra", Item::Value(true.into()));
-    array_of_tables.push(sierra);
-    contract.insert("starknet-contract", Item::ArrayOfTables(array_of_tables));
-
-    document.insert("target", Item::Table(contract));
-}
-
-fn set_cairo_edition(document: &mut DocumentMut, cairo_edition: &str) {
-    document["package"]["edition"] = value(cairo_edition);
-}
-
-fn add_assert_macros(document: &mut DocumentMut, scarb: &Version) -> Result<()> {
-    let version = if scarb < &MINIMAL_SCARB_FOR_CORRESPONDING_ASSERT_MACROS {
-        &DEFAULT_ASSERT_MACROS
-    } else {
-        scarb
-    };
-
-    document
-        .get_mut("dev-dependencies")
-        .and_then(|dep| dep.as_table_mut())
-        .context("Failed to get dev-dependencies from Scarb.toml")?
-        .insert("assert_macros", value(version.to_string()));
-
-    Ok(())
-}
-
-fn extend_gitignore(path: &Path) -> Result<()> {
-    let mut file = OpenOptions::new()
-        .append(true)
-        .open(path.join(".gitignore"))?;
-    writeln!(file, ".snfoundry_cache/")?;
-
-    Ok(())
-}
-
-pub fn run(project_name: &str) -> Result<()> {
+pub fn init(project_name: &str) -> Result<()> {
     let current_dir = std::env::current_dir().context("Failed to get current directory")?;
-    let project_path = current_dir.join(project_name);
-    let scarb_manifest_path = project_path.join("Scarb.toml");
-    let snfoundry_manifest_path = project_path.join("snfoundry.toml");
-
-    // if there is no Scarb.toml run `scarb new`
-    if !scarb_manifest_path.is_file() {
-        ScarbCommand::new_with_stdio()
-            .current_dir(current_dir)
-            .arg("new")
-            .arg(&project_path)
-            .env("SCARB_INIT_TEST_RUNNER", "cairo-test")
-            .run()
-            .context("Failed to initialize a new project")?;
-
-        ScarbCommand::new_with_stdio()
-            .current_dir(&project_path)
-            .manifest_path(scarb_manifest_path.clone())
-            .offline()
-            .arg("remove")
-            .arg("--dev")
-            .arg("cairo_test")
-            .run()
-            .context("Failed to remove cairo_test")?;
-    }
-
-    add_template_to_scarb_manifest(&scarb_manifest_path)?;
-
-    if !snfoundry_manifest_path.is_file() {
-        create_snfoundry_manifest(&snfoundry_manifest_path)?;
+    let project_path = Utf8PathBuf::from_path_buf(current_dir)
+        .expect("Failed to create Utf8PathBuf for the current directory")
+        .join(project_name);
+
+    if !project_path.join("Scarb.toml").exists() {
```

Comment:
Note: This is a hack to prevent printing this warning when running `scarb init/new` with an older version of Scarb.

Reply:
Maybe let's document this in code then